src/destination.rs

```diff
@@ -0,0 +1,382 @@
+use crate::{EndpointsSpec, Error, OverridesSpec};
```

Comment:
This is primarily what was was in `lib.rs`. I moved everything `DstService` related into this file.